### PR TITLE
Client task6

### PR DESCRIPTION
**Pull request by dziq45**

### CODE REVIEW

File: client/src/app/components/page/task-detail/task-detail.component.html [hunk 1]
```diff
@@ -1,18 +1,24 @@
-<div class="card mt-4 cardContainer flex-row">
+<div class="card mt-4 container flex-row">
     <div class="colCard container">
```
Comment (time2funk) [inline client/src/app/components/page/task-detail/task-detail.component.html:2]: you have container inside container
this is not right
please follow the common flow
container -> row -> col

File: server/model/taskModel.js [hunk 2]
```diff
@@ -42,9 +45,8 @@ const TaskModel = {
             description: params.description,
             tasks: [],
             createdBy: params.createdBy,
-            tags: params.tags,
-            priority: params.priority
-
+            tags: params.tags != null? params.tags : [],
```
Comment (time2funk) [inline server/model/taskModel.js:48]: please use strict equal |  not equal operators

File: client/src/app/components/page/task-detail/task-detail.component.ts [hunk 3]
```diff
@@ -158,15 +167,16 @@ export class TaskDetailComponent implements OnInit, OnDestroy {
...
-                task => {
-                    this.task = task;
-                    this.tags = task.tags;
-                    this.taskForm.patchValue(task);
+                ).subscribe(
+                    task => {
+                        this.task = task;
+                        this.tags = task.tags;
+                        this.taskForm.patchValue(task);
+                        this.counterComponent.setDate(new Date(task.deadline));
```
Comment (time2funk) [inline client/src/app/components/page/task-detail/task-detail.component.ts:179]: what if there is no task.deadline ?

File: client/src/app/components/page/task-detail/task-detail.component.ts [hunk 4]
```diff
@@ -68,12 +74,14 @@ export class TaskDetailComponent implements OnInit, OnDestroy {
...
                 parentId: null,
-                tags: this.taskForm.value.tags
+                tags: this.taskForm.value.tags,
+                deadline: this.taskForm.value.deadline
             }
             this.loaderService.setLoading(true);
             this.ts.addTask(newTask).pipe(
                 finalize(() => {
                     this.loaderService.setLoading(false);
+                    this.fetchedData = true;
```
Comment (time2funk) [inline client/src/app/components/page/task-detail/task-detail.component.ts:84]: ? what this is for

File: client/src/app/components/deadline-counter/deadline-counter.component.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+@Component({
+    selector: 'app-deadline-counter',
+    templateUrl: './deadline-counter.component.html',
+    styleUrls: ['./deadline-counter.component.scss']
+})
+export class DeadlineCounterComponent implements OnInit {
+
+    @Input() date: Date;
+    daysLeft: Number;
```
Comment (time2funk) [inline client/src/app/components/deadline-counter/deadline-counter.component.ts:11]: public | private

File: client/src/app/components/date-picker/date-picker.component.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+@Component({
+    selector: 'app-date-picker',
+    templateUrl: './date-picker.component.html',
+    styleUrls: ['./date-picker.component.scss']
+})
+export class DatePickerComponent implements OnInit {
+    @Output() onDatePicked: EventEmitter<Date> = new EventEmitter<Date>();
+    @Input() pickedDate: Date;
+    date: FormControl;
+    time: NgbTimeStruct = { hour: 13, minute: 30, second: 0 };
```
[the remark this answers is not in the document]
Comment (dziq45) [reply client/src/app/components/date-picker/date-picker.component.ts]: Removed

File: client/src/app/components/date-picker/date-picker.component.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+})
+export class DatePickerComponent implements OnInit {
+    @Output() onDatePicked: EventEmitter<Date> = new EventEmitter<Date>();
+    @Input() pickedDate:Date;
+    date:FormControl;
+    constructor() { }
+    ngOnInit() {
+        this.date = new FormControl(this.pickedDate)
+    }
+    public onDateChange(type: string, event: MatDatepickerInputEvent<Date>) {
```
[the remark this answers is not in the document]
Comment (dziq45) [reply client/src/app/components/date-picker/date-picker.component.ts]: Deleted parameter

File: client/src/app/components/page/task-detail/task-detail.component.html (hunk 1 above)
Comment (dziq45) [reply client/src/app/components/page/task-detail/task-detail.component.html]: Okay, changed

File: server/model/taskModel.js (hunk 2 above)
Comment (dziq45) [reply server/model/taskModel.js]: Done

File: client/src/app/components/page/task-detail/task-detail.component.ts (hunk 3 above)
Comment (dziq45) [reply client/src/app/components/page/task-detail/task-detail.component.ts]: Added if condition

File: client/src/app/components/deadline-counter/deadline-counter.component.ts (hunk 5 above)
Comment (dziq45) [reply client/src/app/components/deadline-counter/deadline-counter.component.ts]: Done

File: client/src/app/components/page/task-detail/task-detail.component.ts (hunk 4 above)
Comment (dziq45) [reply client/src/app/components/page/task-detail/task-detail.component.ts:84]: I use it here
```
<app-date-picker *ngIf="fetchedData || !updateMode" [pickedDate]="taskForm.value.deadline"
                (onDatePicked)="onDateChange($event)"></app-date-picker>
```
to pass value after fetching from the server

File: client/src/app/components/date-picker/date-picker.component.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import { NgbTimepickerConfig } from '@ng-bootstrap/ng-bootstrap';
+@Component({
+    selector: 'app-date-picker',
+    templateUrl: './date-picker.component.html',
+    styleUrls: ['./date-picker.component.scss']
+})
+export class DatePickerComponent implements OnInit {
+    @Output() onDatePicked: EventEmitter<Date> = new EventEmitter<Date>();
+    @Input() pickedDate: Date;
+    date: FormControl;
```
Comment (time2funk) [inline client/src/app/components/date-picker/date-picker.component.ts:14]: public private

File: client/src/app/components/page/task-detail/task-detail.component.ts (hunk 4 above)
Comment (time2funk) [reply client/src/app/components/page/task-detail/task-detail.component.ts:84]: please left TODO: refactor this

File: client/src/app/components/date-picker/date-picker.component.ts (hunk 8 above)
Comment (dziq45) [reply client/src/app/components/date-picker/date-picker.component.ts]: Done

File: client/src/app/components/page/task-detail/task-detail.component.ts (hunk 4 above)
Comment (dziq45) [reply client/src/app/components/page/task-detail/task-detail.component.ts:87]: Done

**Pull request merged**